tests/integration/data/datasets.py

```diff
@@ -143,8 +143,28 @@ def org_annotate_dataset(indico):
 def org_annotate_workflow(indico, org_annotate_dataset: Dataset):
     client = IndicoClient()
     workflowreq = CreateWorkflow(dataset_id=org_annotate_dataset.id, name=f"OrgAnnotate-test-{int(time.time())}")
-    response = client.call(workflowreq)
-
+    wf = client.call(workflowreq)
+
+    # TODO: Replace this with AddComponent once it supports outputs
+    # currently on dev, json output component is blueprint 16
+    blueprint_id=16
```

Comment:
blueprintId is optional for certain component types. rather than hard-coding, you can just specify component_type in your config as you already did and this should work

Reply:
or actually, you should just be able to use _AddWorkflowComponent()

Reply:
when does the blueprint id need to be specified?

Reply:
CUSTOM_MODEL and EXTERNAL_MODEL component types require it. optional otherwise (for backwards-compat with 5.X)

Reply:
:+1: dope, thanks for the info